src/ListNavigationEnabled.h

```diff
@@ -0,0 +1,115 @@
+/*!********************************************************************
+
+   Audacity: A Digital Audio Editor
+
+   @file ListNavigationEnabled.h
+
+   @author Vitaly Sverchinsky
+
+**********************************************************************/
+
+#pragma once
+
+#include <wx/containr.h>
+
+/**
+ * \brief Changes default arrow navigation to behave more list- or table-like.
+ * Instead of searching focusable items among children first, list navigation
+ * searches for siblings when arrow key is pressed. Tab behaviour stays same.
+ * Requires wxWANT_CHARS style flag to be set
+ */
+template<class WindowBase>
+class ListNavigationEnabled : public wxNavigationEnabled<WindowBase>
```

Comment:
If I wrote this, I would do the extra work to define a .cpp file, so that (1) at least one .cpp file in the system includes the header file before any other, guaranteeing that the header is self-sufficient for compilation; and (2) making out-of-line functions doing most of the work that doesn't depend on the template type parameter, leaving only small template functions in the header to be inlined.

See for instance Observer, which separates an out-of-line, "type erased" part.

I don't insist that you follow the suggestion for this PR.

Reply:
It would also make sense to put this file in src/wrappers.  The resembles `wxPanelWrapper` for instance.  But I don't insist.  Library extractions might soon upset the folder organization anyway.  Although a library for src/wrappers would definitely be a non-toolkit-neutral part of the program until we figure out better abstractions.